nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1262,6 +1277,96 @@ public void run() {
         }
     }
 
+    protected boolean registerRemoteInstance() throws IOException, ConfigurationException, PackageException {
+        String username;
+        String password;
+        String project;
+        String description = "";
+        NuxeoClientInstanceType type;
+        if (System.console() != null) {
+            username = LauncherRegisterPromptHelper.promptUsername();
+            password = LauncherRegisterPromptHelper.promptPassword();
+            List<ConnectProject> projs = getConnectRegistrationBroker().getAvailableProjects(username, password);
+            project = LauncherRegisterPromptHelper.promptProjectId(projs);
+            type = LauncherRegisterPromptHelper.promptInstanceType();
+            description = LauncherRegisterPromptHelper.promptDescription();
+        } else {
+            String[] input = new String(IOUtils.toByteArray(System.in)).split("\n");
+            if (input.length < 4) {
+                throw new IOException("Wrong number of arguments.");
+            }
+            username = input[0];
+            password = input[1];
+            project = input[2];
+            type = NuxeoClientInstanceType.fromString(input[3]);
+            if (input.length > 3) {
+                description = StringUtils.join(Arrays.copyOfRange(input, 4, input.length), "\n");
+            }
+        }
```

Comment:
This case (the else block) must be described in the usage, it cannot be guessed.
Anyway, I'm wondering why not rather do like in other similar nuxeoctl commands: 
```
if username not set (passed as parameter, option or context) {
    if console !=null {
        read username from console
    } else {
        read username from stdin
}
and so on
```
Refactor that code up to the prompt method and don't bother with the console in this method.